patches/react-native-reanimated+2.8.0.patch

```diff
@@ -0,0 +1,1291 @@
++
+     inOut(easing: Animated.EasingNodeFunction): Animated.EasingNodeFunction;
+   }
+ 
+@@ -1059,27 +1343,37 @@ declare module 'react-native-reanimated' {
+     ease: Animated.EasingFunction;
+     quad: Animated.EasingFunction;
+     cubic: Animated.EasingFunction;
++
+     poly(n: number): Animated.EasingFunction;
++
+     sin: Animated.EasingFunction;
+     circle: Animated.EasingFunction;
+     exp: Animated.EasingFunction;
++
+     elastic(bounciness?: number): Animated.EasingFunction;
++
+     back(s?: number): Animated.EasingFunction;
++
+     bounce: Animated.EasingFunction;
++
+     bezier(
+       x1: number,
+       y1: number,
+       x2: number,
+-      y2: number
++      y2: number,
+     ): { factory: () => Animated.EasingFunction };
++
+     bezierFn(
+       x1: number,
+       y1: number,
+       x2: number,
+-      y2: number
++      y2: number,
+     ): Animated.EasingFunction;
++
+     in(easing: Animated.EasingFunction): Animated.EasingFunction;
++
+     out(easing: Animated.EasingFunction): Animated.EasingFunction;
++
+     inOut(easing: Animated.EasingFunction): Animated.EasingFunction;
+   }
+ 
+@@ -1113,6 +1407,7 @@ declare module 'react-native-reanimated' {
+       | 'slide-right'
+       | 'slide-left';
+   }
++
+   export class Transition extends Component {
+     static In: ComponentClass<TransitionInOutProps>;
+     static Out: ComponentClass<TransitionInOutProps>;
+@@ -1121,11 +1416,15 @@ declare module 'react-native-reanimated' {
+     static Sequence: ComponentClass<{}>;
+   }
+ 
+-  export class Clock extends Animated.Clock {}
+-  export class Value<
+-    T extends string | number | boolean
+-  > extends Animated.Value<T> {}
+-  export class Node<T> extends Animated.Node<T> {}
++  export class Clock extends Animated.Clock {
++  }
++
++  export class Value<T extends string | number | boolean> extends Animated.Value<T> {
++  }
++
++  export class Node<T> extends Animated.Node<T> {
++  }
++
+   export const add: typeof Animated.add;
+   export const sub: typeof Animated.sub;
+   export const multiply: typeof Animated.multiply;
+@@ -1183,5 +1482,19 @@ declare module 'react-native-reanimated' {
+   export const SpringUtils: typeof Animated.SpringUtils;
+   export const useValue: typeof Animated.useValue;
+   export const ReverseAnimation: typeof Animated.ReverseAnimation;
++
+   export function enableLayoutAnimations(flag: boolean): void;
++
++  export class WorkletEventHandler<T> {
++    worklet: (event: T) => void;
++    eventNames: string[];
++    reattachNeeded: boolean;
++    listeners: Record<string, (event: T) => void>;
++    viewTag: number | undefined;
++    registrations: string[];
++    constructor(worklet: (event: T) => void, eventNames?: string[]);
++    updateWorklet(newWorklet: (event: T) => void): void;
++    registerForEvents(viewTag: number, fallbackEventName?: string): void;
++    unregisterFromEvents(): void;
++  }
+ }
```

Comment:
The patch is quite big (1.2k lines). Can you explain why are all this changes needed?

Reply:
I can't need to take a look at it, the only thing that should be changed is adding one class from previous patch. I will fix it, thanks for spotting that!

Reply:
I think I ran code formatter 🤦